src/Data/HashTable/ST/Basic.hs

```diff
@@ -620,6 +681,12 @@ recordIsDeleted = (== deletedMarker)
 
 
 ------------------------------------------------------------------------------
+{-# INLINE recordIsFilled #-}
+recordIsFilled :: Elem -> Bool
+recordIsFilled el = not (recordIsEmpty el) && not (recordIsDeleted el)
```

Comment:
Perhaps something like this would be ok?
```haskell
recordIsFilled !el = isTrue# filled#
  where
    !(I# i#) = fromIntegral el :: Int
    !filled# = (i# /=# 0#) `andI#` (i# /=# 1#)
```
Using `mask` wouldn't give additional improvement here, or am I missing something?